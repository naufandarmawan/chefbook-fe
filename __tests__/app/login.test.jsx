import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import LoginPage from '../../app/(auth)/login/page'
import LoginLayout from '../../app/(auth)/layout'

describe('Login Page', () => {
  it('should render', () => {
    const page = render(
      <LoginLayout>
        <LoginPage />
      </LoginLayout>
    )

    expect(page).toMatchSnapshot()
  })
})