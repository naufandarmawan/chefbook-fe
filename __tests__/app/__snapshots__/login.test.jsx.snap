// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex max-lg:flex-col"
      >
        <div
          class="w-1/2 flex flex-col justify-center items-center bg-cover max-lg:hidden max-lg:h-0"
          style="background-image: url(/auth-background.png);"
        >
          <a
            href="/"
          >
            <img
              class="w-32"
              src="/logo.svg"
            />
          </a>
        </div>
        <div
          class="w-1/2 flex flex-col justify-center items-center py-20 max-lg:w-full max-lg:p-4 max-lg:py-10"
        >
          <div
            class="w-1/2 flex flex-col gap-6 max-lg:w-full"
          >
            <p
              class="text-center font-bold text-3xl text-[#EFC81A]"
            >
              Welcome
            </p>
            <p
              class="text-center font-normal text-lg text-[#8692A6]"
            >
              Log in into your exiting account
            </p>
            <div
              class="flex flex-col gap-4 w-full"
            >
              <label
                class="form-control w-full"
              >
                <div
                  class="label"
                >
                  <span
                    class="label-text"
                  >
                    Email
                  </span>
                </div>
                <label
                  class="input input-bordered w-full flex items-center gap-2 focus-within:!outline-yellow-400"
                  value=""
                >
                  <input
                    class="grow"
                    placeholder="Enter email address"
                    type="email"
                    value=""
                  />
                </label>
              </label>
              <label
                class="form-control w-full"
              >
                <div
                  class="label"
                >
                  <span
                    class="label-text"
                  >
                    Password
                  </span>
                </div>
                <label
                  class="input input-bordered w-full flex items-center gap-2 focus-within:!outline-yellow-400"
                  value=""
                >
                  <input
                    class="grow"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </label>
              </label>
              <div
                class="form-control"
              >
                <label
                  class="label cursor-pointer justify-start gap-4"
                >
                  <input
                    class="checkbox border-yellow-400 checked:border-yellow-500 [--chkbg:theme(colors.yellow.400)] [--chkfg:white]"
                    type="checkbox"
                  />
                  <span
                    class="label-text"
                  >
                    I agree to terms & conditions
                  </span>
                </label>
              </div>
            </div>
            <div
              class="flex flex-col gap-4 items-end"
            >
              <button
                class="btn bg-yellow-400 text-white hover:bg-yellow-500 w-full"
              >
                Log in
              </button>
              <a
                class="font-medium text-sm text-[#999999]"
                href="/forgot-password"
              >
                Forgot Password?
              </a>
            </div>
            <p
              class="text-center text-sm font-medium text-[#999999]"
            >
              Don’t have an account? 
              <a
                class="text-[#EFC81A]"
                href="/register"
              >
                Sign Up
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex max-lg:flex-col"
    >
      <div
        class="w-1/2 flex flex-col justify-center items-center bg-cover max-lg:hidden max-lg:h-0"
        style="background-image: url(/auth-background.png);"
      >
        <a
          href="/"
        >
          <img
            class="w-32"
            src="/logo.svg"
          />
        </a>
      </div>
      <div
        class="w-1/2 flex flex-col justify-center items-center py-20 max-lg:w-full max-lg:p-4 max-lg:py-10"
      >
        <div
          class="w-1/2 flex flex-col gap-6 max-lg:w-full"
        >
          <p
            class="text-center font-bold text-3xl text-[#EFC81A]"
          >
            Welcome
          </p>
          <p
            class="text-center font-normal text-lg text-[#8692A6]"
          >
            Log in into your exiting account
          </p>
          <div
            class="flex flex-col gap-4 w-full"
          >
            <label
              class="form-control w-full"
            >
              <div
                class="label"
              >
                <span
                  class="label-text"
                >
                  Email
                </span>
              </div>
              <label
                class="input input-bordered w-full flex items-center gap-2 focus-within:!outline-yellow-400"
                value=""
              >
                <input
                  class="grow"
                  placeholder="Enter email address"
                  type="email"
                  value=""
                />
              </label>
            </label>
            <label
              class="form-control w-full"
            >
              <div
                class="label"
              >
                <span
                  class="label-text"
                >
                  Password
                </span>
              </div>
              <label
                class="input input-bordered w-full flex items-center gap-2 focus-within:!outline-yellow-400"
                value=""
              >
                <input
                  class="grow"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </label>
            </label>
            <div
              class="form-control"
            >
              <label
                class="label cursor-pointer justify-start gap-4"
              >
                <input
                  class="checkbox border-yellow-400 checked:border-yellow-500 [--chkbg:theme(colors.yellow.400)] [--chkfg:white]"
                  type="checkbox"
                />
                <span
                  class="label-text"
                >
                  I agree to terms & conditions
                </span>
              </label>
            </div>
          </div>
          <div
            class="flex flex-col gap-4 items-end"
          >
            <button
              class="btn bg-yellow-400 text-white hover:bg-yellow-500 w-full"
            >
              Log in
            </button>
            <a
              class="font-medium text-sm text-[#999999]"
              href="/forgot-password"
            >
              Forgot Password?
            </a>
          </div>
          <p
            class="text-center text-sm font-medium text-[#999999]"
          >
            Don’t have an account? 
            <a
              class="text-[#EFC81A]"
              href="/register"
            >
              Sign Up
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
